QtCollider/QcApplication.cpp

```diff
@@ -181,12 +181,27 @@ bool QcApplication::notify(QObject* object, QEvent* event) {
     bool result = QApplication::notify(object, event);
 
 #ifdef Q_OS_MAC
+    bool isCloseEvent = false;
+    bool isKeyEvent = false;
+    if (event->type() == QEvent::KeyPress) {
+        QKeyEvent* kevent = static_cast<QKeyEvent*>(event);
+        if ((kevent->key() != Qt::Key_unknown)) {
+            isKeyEvent = true; // true for regular keys pressed, but not modifiers alone
+        }
+        if ((kevent->key() == Qt::Key_W) && (kevent->modifiers() == Qt::ControlModifier)) {
+            isCloseEvent = true;
+        }
+    }
```

Comment:
you can remove the local variables and place all this logic inside the `if` below, since it only matters `if(result)`. you can also condense these two checks into a single `if`. i am guessing you split it to make this clearer, but in my opinion keeping the logic condensed and having an explanatory comment is much nicer:

```
if (result and key press type) {
  static cast
  if (is not close and is not unknown) {
    accept()
  }
}
```